src/core/namespace.c

```diff
@@ -3291,6 +3309,518 @@ int open_shareable_ns_path(int ns_storage_socket[static 2], const char *path, un
         return 1;
 }
 
+static int is_extension_overlay(const char *path, int fd) {
+        _cleanup_free_ char *source = NULL;
+        _cleanup_close_ int dfd = -EBADF;
+        int r;
+
+        assert(path);
+
+        if (fd < 0) {
+                dfd = chase_and_open(path, /* root= */ NULL, CHASE_TRAIL_SLASH,  /* open_flags= */ 0, /* ret_path= */ NULL);
```

Comment:
is it really necessary to open() the thing? wouldn't an O_PATH fd suffice? i.e. as in simple chase()? (the answer might well be that we need a real thing, just asking).

might add CHASE_MUST_BE_DIRECTORY btw. and/or O_DIRECTORY